cvat/apps/dataset_manager/formats/transformations.py

```diff
@@ -36,6 +36,28 @@ def transform_item(self, item):
 
         return item.wrap(annotations=annotations)
 
+class CVATRleToCOCORle(ItemTransform):
+    @staticmethod
+    def convert_mask(mask, img_h, img_w):
+        rle = mask.points[:-4]
+        left, top, right = list(math.trunc(v) for v in mask.points[-4:-1])
+        mat = np.zeros((img_h, img_w), dtype=np.uint8)
+        width = right - left + 1
+        value = 0
+        offset = 0
+        for rleCount in rle:
+            rleCount = math.trunc(rleCount)
+            while rleCount > 0:
+                x, y = offset % width, offset // width
+                mat[y + top][x + left] = value
+                rleCount -= 1
+                offset += 1
+            value = abs(value - 1)
+
+        rle = mask_utils.encode(np.asfortranarray(mat))
```

Comment:
There is an optimized version of mask to RLE conversion in Datumaro: https://github.com/cvat-ai/datumaro/blob/develop/datumaro/util/mask_tools.py#L155

Reply:
<img width="182" alt="image" src="https://user-images.githubusercontent.com/40690378/196116189-578b4cdd-f7e4-44bf-a8e1-2153ea1da004.png">

Optimized?)

Reply:
Probably, it may depend on the hardware or library versions, because when I did this, it was like the opposite.

Reply:
@zhiltsov-max 

I believe need to investigate this one. If for some users it works 10 time slower than for others.

I can provide my specs:

- Intel i7 12700K
- Python 3.8.10
- Ubuntu 20.04 (Run in VMWare Player 16)
- Pip freeze:

<details>
```
absl-py==1.3.0
asgiref==3.5.2
astroid==2.11.7
astunparse==1.6.3
attrs==21.4.0
av==8.0.2
azure-core==1.26.0
azure-storage-blob==12.13.0
boto3==1.17.61
botocore==1.20.112
cachetools==5.2.0
certifi==2022.9.24
cffi==1.15.1
charset-normalizer==2.0.12
click==7.1.2
contourpy==1.0.5
coreapi==2.3.3
coreschema==0.0.4
croniter==1.3.7
cryptography==38.0.1
cycler==0.11.0
datumaro==0.3
defusedxml==0.7.1
Deprecated==1.2.13
dill==0.3.5.1
diskcache==5.0.2
dj-pagination==2.5.0
dj-rest-auth==2.2.4
Django==3.2.15
django-allauth==0.50.0
django-appconf==1.0.4
django-auth-ldap==2.2.0
django-compressor==2.4
django-cors-headers==3.5.0
django-extensions==3.0.8
django-filter==2.4.0
django-rq==2.3.2
django-sendfile2==0.7.0
djangorestframework==3.12.4
dnspython==2.2.0
drf-spectacular==0.22.1
EasyProcess==0.3
entrypoint2==1.1
flatbuffers==22.9.24
fonttools==4.37.4
furl==2.1.0
gast==0.5.3
gitdb==4.0.9
GitPython==3.1.27
google-api-core==2.10.2
google-auth==2.12.0
google-auth-oauthlib==0.4.6
google-cloud-core==2.3.2
google-cloud-storage==1.42.0
google-crc32c==1.5.0
google-pasta==0.2.0
google-resumable-media==2.4.0
googleapis-common-protos==1.56.4
grpcio==1.49.1
h5py==3.6.0
idna==3.4
importlib-resources==5.10.0
inflection==0.5.1
isodate==0.6.1
isort==5.10.1
itypes==1.2.0
Jinja2==3.1.2
jmespath==0.10.0
jsonschema==4.16.0
keras==2.8.0
Keras-Preprocessing==1.1.2
kiwisolver==1.4.4
lazy-object-proxy==1.7.1
libclang==14.0.6
limits==2.7.0
lxml==4.9.1
Markdown==3.2.2
MarkupSafe==2.1.1
matplotlib==3.6.1
mccabe==0.7.0
mistune==2.0.4
msrest==0.7.1
natsort==8.0.0
networkx==2.8.7
nibabel==4.0.2
numpy==1.22.0
oauthlib==3.2.1
opencv-python==4.6.0.66
opencv-python-headless==4.5.5.62
opt-einsum==3.3.0
orderedmultidict==1.0.1
orjson==3.8.0
packaging==21.3
pandas==1.5.0
patool==1.12
pdf2image==1.14.0
Pillow==9.0.1
pkgutil-resolve-name==1.3.10
platformdirs==2.5.2
protobuf==4.21.7
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycocotools==2.0.5
pycparser==2.21
Pygments==2.7.4
PyJWT==2.5.0
pylint==2.14.5
pylint-django==2.5.3
pylint-plugin-utils==0.7
pylogbeat==2.0.0
pyparsing==3.0.9
pyrsistent==0.18.1
python-dateutil==2.8.2
python-ldap==3.4.0
python-logstash-async==2.2.0
python3-openid==3.2.0
pytz==2020.1
pyunpack==0.2.1
PyYAML==6.0
rcssmin==1.0.6
redis==3.5.3
requests==2.26.0
requests-oauthlib==1.3.1
rjsmin==1.1.0
rope==0.17.0
rq==1.5.1
rq-scheduler==0.10.0
rsa==4.9
ruamel.yaml==0.17.21
ruamel.yaml.clib==0.2.6
rules==2.2
s3transfer==0.4.2
scipy==1.9.2
Shapely==1.7.1
six==1.16.0
smmap==5.0.0
snakeviz==2.1.0
sqlparse==0.4.2
tensorboard==2.8.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorboardX==2.5.1
tensorflow==2.8.1
tensorflow-estimator==2.8.0
tensorflow-io-gcs-filesystem==0.27.0
termcolor==2.0.1
tomli==2.0.1
tomlkit==0.11.5
tornado==6.2
types-cryptography==3.3.23
typing-extensions==4.4.0
uritemplate==4.1.1
urllib3==1.26.12
Werkzeug==1.0.1
wrapt==1.14.1
zipp==3.9.0
```
</details>

Reply:
Could you please create an issue in the datumaro repo? I'm not sure we have resources for this right now, but it looks like an important task which should be done. For now, let's take your version.